### PR TITLE
Modify the image path of calico for supporting multi-arch

### PR DESCRIPTION
**Pull request by Jingzhao123**
Modify the image path of _config.yml for auto generating
multi-arch yaml configuration files.

Signed-off-by: Jingzhao Ni <jingzhao.ni@arm.com>

## Todos

- [x] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -19,11 +19,11 @@ noderunning: calico-node
...
-  cni: quay.io/calico/cni
-  kubeControllers: quay.io/calico/kube-controllers
+  node: calico/node
+  calicoctl: calico/ctl
+  typha: calico/typha
+  cni: calico/cni
+  kubeControllers: calico/kube-controllers
   calico-upgrade: quay.io/calico/upgrade
   flannel: quay.io/coreos/flannel
   dikastes: quay.io/calico/dikastes
```
Comment (caseydavenport) [inline _config.yml]: I think we need to pull these last three from dockerhub as well.

**Comment by caseydavenport** [comment]
@Jingzhao123 thanks for the PR! I've just added one comment - I think there are a few more images with multi-arch support we'll need to switch to dockerhub.

**Comment by Jingzhao123** [comment]
> Looking at this again, I think we may need to hold off on this until we adjust some infrastructure to handle it.
> 
> Since `_config.yaml` is a global file, this will impact _all_ releases, not just the current one. We have plans to fix this by moving the images names into per-release configuration, but until that's done I don't think we can merge this since it would modify all previous releases.

**Comment by Jingzhao123** [comment]
> Looking at this again, I think we may need to hold off on this until we adjust some infrastructure to handle it.
> 
> Since `_config.yaml` is a global file, this will impact _all_ releases, not just the current one. We have plans to fix this by moving the images names into per-release configuration, but until that's done I don't think we can merge this since it would modify all previous releases.

After thinking it carefully, i agree with you.  I think it is not a good way to change it. Can i add a building parameter for specific architecture. After adding this, we could support the different cpu architect.

**Comment by Jingzhao123** [comment]
> Looking at this again, I think we may need to hold off on this until we adjust some infrastructure to handle it.
> 
> Since `_config.yaml` is a global file, this will impact _all_ releases, not just the current one. We have plans to fix this by moving the images names into per-release configuration, but until that's done I don't think we can merge this since it would modify all previous releases.

OK, thanks. It is a good news. Is there anything i can help?

**Comment by Jingzhao123** [comment]
> Looking at this again, I think we may need to hold off on this until we adjust some infrastructure to handle it.
> 
> Since `_config.yaml` is a global file, this will impact _all_ releases, not just the current one. We have plans to fix this by moving the images names into per-release configuration, but until that's done I don't think we can merge this since it would modify all previous releases.

Hi, caseydavenport,
          Is there any plan about it? And Could you please share it with me, if has a plan ?

**Pull request closed without merging**